src/main/java/duckie/task/Task.java

```diff
@@ -0,0 +1,72 @@
+package duckie.task;
+
+/**
+ * Parent class of all Tasks type
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Instantiate a Task object
+     * @param description Description of the task
+     */
+    public Task(String description) {
+        this.description = description;
+        isDone = false;
+    }
+
+    /**
+     * Check the task and mark as done
+     */
+    public void markDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Check if a Task is completed or not
+     * @return Boolean value indicating the completeness of the task
+     */
+    public boolean isCompleted() {
+        return this.isDone;
+    }
+
+    /**
+     * Return the description of a Task
+     * @return String description of a Task
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Return the type of a Task
+     * @return null
+     */
+    public String getType() {
```

Comment:
Perhaps you may consider change Task to an abstract class so that the implementation of getType(), getDate() can added later in child classes, instead of returning null value.